lib/ssm_config.rb

```diff
@@ -1,39 +1,132 @@
 class SsmConfig
   VERSION = '0.1.1'.freeze
   CONFIG_PATH = 'config'.freeze
-
+  TABLE_NAME = 'ssm_config_records'.freeze
   class << self
-    def method_missing(meth, *args, &block)
+    def method_missing(meth, args = '')
       config_file = Rails.root.join(CONFIG_PATH, "#{meth}.yml")
+      # specify certain model name?
+      if ActiveRecord::Base.connection.table_exists? 'ssm_config_records'
+        puts("hello")
+
+        match_file = SsmConfigDummy.where(:file => meth.to_s)
+        match_key = match_file.where('accessor_keys LIKE :prefix', :prefix => "#{args}%")
 
+        hashes = {}
+
+        match_key.each do |row|
+          hashes[row.accessor_keys] = row.value
+        end
+        puts(hashes)
+
+        hashes = hashes.sort.to_h
+
+        reconstructed = clean_hash(reconstruct_hash(hashes))
+        return access_with_hashkey(reconstructed, args)
```

Comment:
I think all of this should be extracted into a class called `SsmStorage`
``` ruby
 class SssmStorageFile
       def initialize(file_name, *args)
       end
 end

 class SsmDatabaseStorage
    def initialize(file_name, *args)
    end
 end
```

Based on if the table exists or not, we instantiate either or class

Reply:
The implementation of extracting hash should be inside each class